src/libs/actions/User.js

```diff
@@ -290,6 +259,61 @@ function clearContactMethodErrors(contactMethod, fieldName) {
     });
 }
 
+/**
+ * Adds a secondary login to a user's account
+ *
+ * @param {String} contactMethod
+ * @param {String} password
+ */
+function addNewContactMethodAndNavigate(contactMethod, password) {
+    const optimisticData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                partnerUserID: contactMethod,
+                validatedDate: '',
+                errorFields: {
+                    addedLogin: null,
+                },
+                pendingFields: {
+                    addedLogin: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+    }];
+    const successData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                pendingFields: {
+                    addedLogin: null,
+                },
+            },
+        },
+    }];
+    const failureData = [{
+        onyxMethod: CONST.ONYX.METHOD.MERGE,
+        key: ONYXKEYS.LOGIN_LIST,
+        value: {
+            [contactMethod]: {
+                errorFields: {
+                    addedLogin: {
+                        [DateUtils.getMicroseconds()]: Localize.translateLocal('contacts.genericFailureMessages.addContactMethod'),
+                    },
+                },
+                pendingFields: {
+                    addedLogin: null,
+                },
+            },
+        },
+    }];
+
+    API.write('AddNewContactMethod', {partnerUserID: contactMethod, password}, {optimisticData, successData, failureData});
```

Comment:
Not lowercasing the `contactMethod` caused issues when people started testing mixed-case contact methods - https://github.com/Expensify/App/issues/19080